<?xml version="1.0"?>
<launch>

    <!-- Configurable options -->
    <arg name="robot_name"/> <!-- Robot name defines the namespace for all topics regarding the robot -->
    <arg name="urdf_file"/>
    <arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
    <arg name="pos_x" default="0.0"/>
    <arg name="pos_y" default="0.0"/>
    <arg name="pos_z" default="1.0"/>
    <arg name="id" default="1.0"/>

   <arg name="use_ground_truth_for_tf" default="true" />
   <arg name="use_ground_truth_for_control" default="true" />

    <!-- Spawn a object: send the robot XML to param server, push robot_description to factory and spawn robot in gazebo-->
    <arg name="model" default="$(find mbzirc_gcs_view)/urdf/$(arg urdf_file)"/>
    <param name="robot_description_$(arg robot_name)" command="$(find xacro)/xacro.py '$(arg model)' model_name:=$(arg robot_name) id:=$(arg id)" />
    <param name="tf_prefix" type="string" value="$(arg tf_prefix)" />

    <node name="spawn_$(arg robot_name)" pkg="gazebo_ros" type="spawn_model"
        args="-param robot_description_$(arg robot_name) -urdf -x $(arg pos_x) -y $(arg pos_y) -z $(arg pos_z) -model $(arg robot_name)"
        respawn="false">
    </node>

   <!-- START robot state publisher -->
   <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher_$(arg robot_name)" output="screen" >
        <remap from="robot_description" to="robot_description_$(arg robot_name)" />
        <remap from="joint_states" to="joint_states_$(arg robot_name)" />
        <param name="publish_frequency" type="double" value="50.0" />
   </node>

   <!-- PUBLISH state and tf -->
   <!--node name="ground_truth_to_tf_$(arg robot_name)" pkg="message_to_tf" type="message_to_tf" output="screen">
     <param name="odometry_topic" value="/$(arg robot_name)/ground_truth/state" />
     <param name="frame_id" value="map" />
     <param name="footprint_frame_id" value="base_footprint_$(arg id)" />
     <param name="stabilized_frame_id" value="base_stabilized_$(arg id)" />
     <param name="child_frame_id" value="base_link_$(arg id)" />
     <param name="tf_prefix" value="$(arg tf_prefix)" if="$(arg use_ground_truth_for_tf)" />
     <param name="tf_prefix" value="$(arg tf_prefix)/ground_truth" unless="$(arg use_ground_truth_for_tf)" />
   </node-->

    <!--node name="grvc_hal_$(arg robot_name)" pkg="grvc_quadrotor_hal" type="hal_node" respawn="false" output="screen"
        args="-hal_ns=/$(arg robot_name)/hal -wp_topic=go_to_wp -gazebo_ns=/$(arg robot_name)"/-->

    <!--node name="take_object_$(arg robot_name)" pkg="mbzirc_gcs_view" type="take_object" respawn="false" output="screen"
        args="-hal_ns=/$(arg robot_name)/hal -wp_topic=go_to_wp -gazebo_ns=/$(arg robot_name) -start_pos=$(arg pos_x),$(arg pos_y),$(arg pos_z)"/-->

</launch>
